src/mlpack/methods/ann/activation_functions/gaussian_function.hpp

```diff
@@ -0,0 +1,82 @@
+/**
+ * @file gaussian_function.hpp
+ * @author Himanshu Pathak
+ *
+ * Definition and implementation of the gaussian function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_GAUSSIAN_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_GAUSSIAN_FUNCTION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The gaussian function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) &=& \frac{1}{e^{-1 * x^2}} \\
+ * f'(x) &=& f(x) * -x * 2)
+ * @f}
+ */
+class GaussianFunction
+{
+ public:
+  /**
+   * Computes the gaussian function.
+   *
+   * @param x Input data.
+   * @return f(x).
+   */
+  template<typename eT>
+  static double Fn(const eT x)
+  {
+    return std::exp(-1 * std::pow(x, 2));
+  }
+
+  /**
+   * Computes the gaussian function.
+   *
+   * @param x Input data.
+   * @param y The resulting output activation.
+   */
+  template<typename InputVecType, typename OutputVecType>
+  static void Fn(const InputVecType& x, OutputVecType& y)
+  {
+    y = arma::exp(-1 * arma::pow(x, 2));
```

Comment:
@saksham189 In the RBFN layer, we are only calculating distance Gaussian is implemented as a separate function.

Reply:
The activation function that is used is part of the `radial_basis_function` so, I think we should use it inside the layer itself. Do you think that you can a template parameter for the activation function?

Reply:
If I implement this inside we will be stuck at using only gaussian inside RBF. I have a different idea of this in my mind when RBF part will complete we can use Activation function inside layer just we are using TANH in LSTM. We can initialize different activation function according to our requirement and create a new parameter `activation` in `RBF` layer to ask it from user. What do you think ??